include/Parser/include/parser.h

```diff
@@ -0,0 +1,151 @@
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <stdexcept>
+
+class JSON {
+public:
+    enum Type { OBJECT, STRING, NUMBER, BOOL, NIL }; // Enum to define the type of JSON value
+
+    JSON() : type(NIL) {} // Default constructor for NIL type
+    JSON(Type t) : type(t) {} // Constructor for different types
+    JSON(const std::string &value) : type(STRING), str_value(value) {} // Constructor for STRING type
+    JSON(double value) : type(NUMBER), num_value(value) {} // Constructor for NUMBER type
+    JSON(bool value) : type(BOOL), bool_value(value) {} // Constructor for BOOL type
+
+    Type get_type() const { return type; } // Getter for the type of JSON value
+
+    const std::unordered_map<std::string, JSON>& get_object_value() const { return object_value; } // Getter for OBJECT value
+    const std::string& get_string_value() const { return str_value; } // Getter for STRING value
+    double get_number_value() const { return num_value; } // Getter for NUMBER value
+    bool get_bool_value() const { return bool_value; } // Getter for BOOL value
+
+    static JSON Parse(const std::string &content); // Static method to parse JSON from string
+
+private:
+    static JSON ParseObject(std::stringstream &ss); // Method to parse JSON object
+    static JSON ParseString(std::stringstream &ss); // Method to parse JSON string
+    static JSON ParseNumber(std::stringstream &ss); // Method to parse JSON number
+    static JSON ParseBool(std::stringstream &ss); // Method to parse JSON boolean
+    static JSON ParseNull(std::stringstream &ss); // Method to parse JSON null
+    static void SkipWhitespace(std::stringstream &ss); // Helper method to skip whitespace characters
+
+    Type type; // Type of the JSON value
+    std::unordered_map<std::string, JSON> object_value; // Holds OBJECT type values
+    std::string str_value; // Holds STRING type values
+    double num_value; // Holds NUMBER type values
+    bool bool_value; // Holds BOOL type values
+};
+
+// Helper method to skip whitespace characters in the stream
+void JSON::SkipWhitespace(std::stringstream &ss) {

```

Comment:
You should separate definition from declaration(or mark function as inline if you want to keep it in one file) because there is no sense to keep it in one translation unit